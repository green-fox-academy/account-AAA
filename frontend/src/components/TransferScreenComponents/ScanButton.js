import React from 'react';
import { Text, Button, Icon } from 'native-base';
import { withNavigation } from 'react-navigation';
import buttonStyle from '../../styles/BottomButtonStyle';
import navigationPropTypes from '../../helpers/navigationPropTypes';

function ScanButton({ navigation }) {
  const navToNextPage = (screenName) => {
    navigation.navigate(screenName);
  };

  return (
    <Button
      style={{ ...buttonStyle.button, alignSelf: 'center' }}
      iconLeft
      info
      onPress={() => navToNextPage('QRCodeScanner')}
    >
      <Icon name="camera" />
      <Text>Scan</Text>
    </Button>
  );
}

ScanButton.propTypes = {
  navigation: navigationPropTypes.isRequired,
};

export default withNavigation(ScanButton);
