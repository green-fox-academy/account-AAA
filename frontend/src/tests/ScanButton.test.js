import React from 'react';
import renderer from 'react-test-renderer';
import ScanButton from '../components/TransferScreenComponents/ScanButton';

describe('<ScanButton />', () => {
  it('render correctly', () => {
    const data = {
      account: {
        id: 1,
        depositName: 'Main Account',
      },
    };
    const navigation = {
      getParam: (key) => data[key],
    };
    const tree = renderer
      .create(
        <ScanButton navigation={navigation} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
