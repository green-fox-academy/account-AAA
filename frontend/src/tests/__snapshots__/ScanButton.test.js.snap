// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScanButton /> render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#62B1F6",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "color": "white",
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "space-around",
      "marginBottom": 50,
      "marginRight": 30,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 120,
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
  >
    Scan
  </Text>
</View>
`;
